### PR TITLE
8329112: Clean up CDS checking of unsupported module options

### PR DESCRIPTION
**Pull request by iklam**
- Renamed the functions to `check_unsupported_dumping_module_options()` and `has_unsupported_runtime_module_options()`, since their error messages are about VM options, not system properties.
- Consolidated the search of unsupported properties into a helper function.
- Removed obsolete code/comment in classLoader.cpp that suggests that `--patch-module` is allowed during dumping. Replace with an assertion that forbids patching when dumping.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8329112](https://bugs.openjdk.org/browse/JDK-8329112): Clean up CDS checking of unsupported module options (**Enhancement** - P4)

### Reviewers
 * [Calvin Cheung](https://openjdk.org/census#ccheung) (@calvinccheung - **Reviewer**) ⚠️ Review applies to [84f470d8](https://git.openjdk.org/jdk/pull/18492/files/84f470d8b81f8fc92eeaf8f80e86ee0fbf4bef59)
 * [Matias Saavedra Silva](https://openjdk.org/census#matsaave) (@matias9927 - Committer) ⚠️ Review applies to [641763ac](https://git.openjdk.org/jdk/pull/18492/files/641763acca7ef551cfdba2dc059e65f546fd8230)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18492/head:pull/18492` \
`$ git checkout pull/18492`

Update a local copy of the PR: \
`$ git checkout pull/18492` \
`$ git pull https://git.openjdk.org/jdk.git pull/18492/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18492`

View PR using the GUI difftool: \
`$ git pr show -t 18492`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18492.diff">https://git.openjdk.org/jdk/pull/18492.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18492#issuecomment-2020956316)

### CODE REVIEW

File: src/hotspot/share/classfile/classLoader.cpp [hunk 1]
```diff
@@ -1141,6 +1131,7 @@ InstanceKlass* ClassLoader::load_class(Symbol* name, PackageEntry* pkg_entry, bo
     } else {
       // Exploded build - attempt to locate class in its defining module's location.
       assert(_exploded_entries != nullptr, "No exploded build entries present");
+      assert(!CDSConfig::is_dumping_archive(), "CDS doesn't support exploded build");
```
Comment (calvinccheung) [inline src/hotspot/share/classfile/classLoader.cpp:1134]: Suggestion: update the assert comment to `"CDS dumping doesn't support exploded build"`.

**Review by calvinccheung — APPROVED** [review]
I've one suggestion. Looks good otherwise.

File: src/hotspot/share/classfile/classLoader.cpp (hunk 1 above)
Comment (iklam) [reply src/hotspot/share/classfile/classLoader.cpp]: Fixed. Thanks for the review.

**Review by matias9927 — APPROVED** [review]
Looks good, thanks!

**Comment by iklam** [comment]
Thanks @calvinccheung and @matias9927 for the review.
/integrate

**Pull request closed without merging**